x-pack/plugins/stack_alerts/public/alert_types/components/data_view_select_popover.tsx

```diff
@@ -14,56 +14,101 @@ import {
-  EuiFormRow,
   EuiPopover,
   EuiPopoverFooter,
   EuiPopoverTitle,
   EuiText,
   useEuiPaddingCSS,
 } from '@elastic/eui';
-import { DataViewsList } from '@kbn/unified-search-plugin/public';
-import { DataViewListItem } from '@kbn/data-views-plugin/public';
-import { useTriggersAndActionsUiDeps } from '../es_query/util';
+import type { DataViewListItem, DataView } from '@kbn/data-views-plugin/public';
+import { useDiscoverAlertContext } from '@kbn/discover-plugin/public';
+import { DataViewSelector } from '@kbn/unified-search-plugin/public';
+import { useDiscoverAlertServices } from '../es_query/util';
 
 export interface DataViewSelectPopoverProps {
-  onSelectDataView: (newDataViewId: string) => void;
-  dataViewName?: string;
-  dataViewId?: string;
+  dataView: DataView;
+  onSelectDataView: (selectedDataView: DataView) => void;
 }
 
+const toDataViewListItem = (dataView: DataView): DataViewListItem => {
+  return {
+    id: dataView.id!,
+    title: dataView.title,
+    name: dataView.name,
+  };
+};
+
 export const DataViewSelectPopover: React.FunctionComponent<DataViewSelectPopoverProps> = ({
+  dataView,
   onSelectDataView,
-  dataViewName,
-  dataViewId,
 }) => {
-  const { data, dataViewEditor } = useTriggersAndActionsUiDeps();
-  const [dataViewItems, setDataViewsItems] = useState<DataViewListItem[]>();
+  const { dataViews, dataViewEditor } = useDiscoverAlertServices();
+  const discoverContext = useDiscoverAlertContext();
+  const [adHocDataViews, setAdHocDataViews] = useState<DataViewListItem[]>(() => {
+    if (discoverContext.isManagementPage) {
+      return dataView.isPersisted() ? [] : [toDataViewListItem(dataView)];
+    }
+    return discoverContext.initialAdHocDataViewList;
+  });
+  const [dataViewItems, setDataViewsItems] = useState<DataViewListItem[]>([]);
   const [dataViewPopoverOpen, setDataViewPopoverOpen] = useState(false);
 
   const closeDataViewEditor = useRef<() => void | undefined>();
 
-  const loadDataViews = useCallback(async () => {
-    const fetchedDataViewItems = await data.dataViews.getIdsWithTitle();
-    setDataViewsItems(fetchedDataViewItems);
-  }, [setDataViewsItems, data.dataViews]);
+  const allDataViewItems = useMemo(
+    () => [...dataViewItems, ...adHocDataViews],
+    [adHocDataViews, dataViewItems]
+  );
 
   const closeDataViewPopover = useCallback(() => setDataViewPopoverOpen(false), []);
 
+  const onChangeDataView = useCallback(
+    async (selectedDataViewId: string) => {
+      const selectedDataView = await dataViews.get(selectedDataViewId);
+      onSelectDataView(selectedDataView);
+      closeDataViewPopover();
+    },
+    [closeDataViewPopover, dataViews, onSelectDataView]
+  );
+
+  const loadPersistedDataViews = useCallback(async () => {
+    const ids = await dataViews.getIds();
+    const dataViewsList = await Promise.all(ids.map((id) => dataViews.get(id)));
+
+    setDataViewsItems(
+      dataViewsList.filter((current) => current.isTimeBased()).map(toDataViewListItem)
```

Comment:
Should we also filter out ad hoc data views without time fields?

Reply:
With this change we could also create adhoc data view from alerts popover with or without time field name (see screenshot). So filtering out will create inconsistency (created data view will not appear in selector).

<img height="200" src="https://user-images.githubusercontent.com/39378793/196912756-33c76e7d-47e8-482d-bf8d-d2e6f78eb731.jpeg" >